Davidfowl/whats new

davidfowl opened the pull request:
This is for review

Preview: https://github.com/dotnet/aspire/blob/davidfowl/whats-new/tools/ReleaseNotes/data/whats-new-13.md

davidfowl committed:
Add 13.0 milestone closed items snapshot (generated summaries)

davidfowl committed:
Add release notes for Aspire 13.0, detailing major features, upgrade instructions, and breaking changes

davidfowl committed:
Enhance Dockerfile Builder API with fluent syntax and multi-stage support

davidfowl committed:
Reorganize "What's New" section in release notes for Aspire 13.0 to improve clarity and accessibility of major features

davidfowl committed:
Add new integration packages and resources for Aspire 13.0, including Vite app support, .NET MAUI integration, and Python enhancements

davidfowl committed:
Revise "What's New" section in release notes for Aspire 13.0 to highlight polyglot platform features, including first-class support for Python and JavaScript, and introduce new integration packages and resources.

davidfowl committed:
Refine Dockerfile generation and certificate trust for Python and JavaScript applications, simplifying configuration and enhancing automatic trust features.

davidfowl committed:
Update release notes for Aspire 13.0 to reflect new documentation site and enhance clarity on certificate trust across languages

davidfowl committed:
Update environment variable configuration for Python module to use URI format

davidfowl committed:
Enhance `aspire update` command with self-update feature and reliability improvements

davidfowl committed:
Revise "What's New" section in release notes for Aspire 13.0 to improve clarity on CLI command updates and introduce the Distributed Application Pipeline features.

davidfowl committed:
Update release notes for Aspire 13.0 to include details on the new `aspire do` command and its role in the Distributed Application Pipeline, enhancing clarity on pipeline execution and CLI improvements.

davidfowl committed:
Update release notes for Aspire 13.0 to include new section on simplified service URL environment variables, enhancing clarity on service discovery for non-.NET applications.

davidfowl committed:
Refactor deployment commands to simplify usage by replacing `aspire do deploy` with `aspire deploy` and introduce `aspire do diagnostics` for enhanced pipeline troubleshooting.

davidfowl committed:
Update release notes for Aspire 13.0 to highlight Azure App Service enhancements, including the new Aspire Dashboard and Application Insights integration for improved monitoring and deployment experience.

eerhardt commented on tools/ReleaseNotes/data/whats-new-13.md:
```diff
@@ -0,0 +1,1790 @@
+
+### Container Files as Build Artifacts
+
+Aspire 13 introduces a fundamental paradigm shift in how build artifacts are handled: **containers are now first-class build artifacts**, not just folders on a file system. This change enables reproducible, isolated, and portable builds that work consistently across development, CI/CD, and production environments.
```
This section talks too much about "what are the benefits of using Docker to build your app". Instead I think it should focus more about the new feature - one resource can build a docker image with files in it. And another resource can "consume" or copy those files into its output.

eerhardt commented on tools/ReleaseNotes/data/whats-new-13.md:
```diff
@@ -0,0 +1,1790 @@
+var builder = DistributedApplication.CreateBuilder(args);
+
+var api = builder.AddProject<Projects.Api>("api")
+    .WithContainerFiles("./config", "/app/config", continueOnError: false)
```
`WithContainerFiles` is actually something different and shouldn't be shown here.

davidfowl committed:
Revise container files section in release notes to clarify the new approach of using containers as first-class build artifacts, enhancing reproducibility and isolation in builds.

davidfowl committed:
Update release notes for Aspire 13.0 to document CLI command changes, including removal of `--watch` flag from `aspire run` and behavior modifications for `EndpointReference.GetValueAsync`.

davidfowl committed:
Add documentation for resetting deployment state with `--clear-cache` flag

davidfowl committed:
Revise release notes to clarify the new foundational pipeline system for build, publish, and deployment, and enhance the description of the modern CLI features.

davidfowl committed:
Enhance database connection management with named references and environment variable injection for improved service discovery and configuration flexibility.

JamesNK commented on tools/ReleaseNotes/data/whats-new-13.md:
```diff
@@ -0,0 +1,1732 @@
+- **Built-in logging**: Use `context.Logger` to log step progress
+- **CLI execution**: Run specific steps with `aspire do <step-name>`
+
+#### Executing pipelines with aspire do
```
```suggestion
#### Executing pipelines with `aspire do`
```

JamesNK commented on tools/ReleaseNotes/data/whats-new-13.md:
```diff
@@ -0,0 +1,1732 @@
+
+## 📊 Dashboard enhancements
+
+### Model Context Protocol (MCP) server
```
```suggestion
### Aspire MCP server
```

davidfowl committed:
Revise release notes to remove sections on console logs refactoring and markdown rendering, streamlining the document for clarity and focus on key features.

davidfowl committed:
Update release notes to reflect changes in API naming conventions and mark obsolete APIs in Aspire 13.0

davidfowl committed:
Revise release notes to update section titles and improve clarity for the new `aspire do` pipeline system, reflecting changes in terminology and structure.

davidfowl committed:
Revise release notes to clarify the description of the `aspire do` platform for build, publish, and deployment pipelines.

davidfowl committed:
Revise release notes to enhance clarity and organization, adding sections for Python version detection and non-interactive mode for CI/CD, while updating existing headings for consistency.

davidfowl committed:
Revise release notes to clarify the `--non-interactive` mode for CI/CD, simplifying the description and removing redundant details for improved readability.

davidfowl committed:
Remove troubleshooting section for Python debugging, container file extraction, and pipeline step failures from release notes to streamline content and improve clarity.

davidfowl committed:
Enhance MCP server integration in the Dashboard, detailing capabilities, getting started instructions, and available tools for AI assistants, improving usability and clarity.

JamesNK commented on tools/ReleaseNotes/data/whats-new-13.md:
```diff
@@ -0,0 +1,1853 @@
+- **Telemetry access**: Retrieve structured logs and distributed traces
+- **Command execution**: Run resource commands through AI assistants
+- **Privacy control**: Exclude sensitive resources from MCP using `ExcludeFromMcp()` annotation
+
```
```suggestion
```

davidfowl committed:
Revise release notes to enhance clarity and organization, updating section headings with appropriate emojis and refining the description of the MCP server integration in the Dashboard.

davidfowl committed:
Update release notes to reflect version 13.0, enhancing clarity and consistency throughout the document.

davidfowl committed:
Revise release notes to clarify C# file-based app support, enhancing accuracy and consistency in documentation.

davidfowl committed:
Revise Python debugging support in release notes for clarity and conciseness, removing redundant details and enhancing readability.

davidfowl committed:
Enhance support for C# file-based applications in Aspire 13.0, enabling seamless integration without full project files and improving debugging capabilities in VS Code.

davidfowl committed:
Enhance dynamic input system by adding support for database type and version selection, enabling context-aware loading of options based on user input.

davidfowl committed:
Refine non-interactive mode documentation for CI/CD, clarifying usage and environment variable settings while removing redundant examples.

davidfowl committed:
Refactor lifecycle hook registration and event subscription methods for improved clarity and functionality in Aspire 13.0.

davidfowl committed:
Add breaking change notice for package rename from Aspire.Hosting.NodeJs to Aspire.Hosting.JavaScript

davidfowl committed:
Update Python support documentation for Aspire 13.0, enhancing clarity on package management options and method names.

davidfowl committed:
Enhance Python package management documentation by adding automatic detection details and clarifying usage instructions for uv and pip configurations.

davidfowl committed:
Fix wording in release notes to clarify Aspire 13.0 as a polyglot application platform

davidfowl committed:
Update Python application examples in release notes for clarity and accuracy

davidfowl committed:
Clarify virtual environment configuration in Python support documentation, specifying default behavior and usage examples for `WithVirtualEnvironment()`.

davidfowl committed:
Add documentation for `aspire new` command with curated starter templates

davidfowl committed:
Update JDBC connection string documentation for security best practices and add MAUI platform support for Android and iOS

danegsta commented on tools/ReleaseNotes/data/whats-new-13.md:
````diff
@@ -0,0 +1,1561 @@
+```
+
+The `AddUvicornApp` method automatically:
+- Configures HTTP/HTTPS endpoints
````
Support for configuring HTTPS endpoints will come in 13.1:
```suggestion
- Configures HTTP endpoints
```

danegsta commented on tools/ReleaseNotes/data/whats-new-13.md:
```diff
@@ -0,0 +1,1561 @@
+
+#### Certificate Trust Across Languages
+
+Aspire 13.0 automatically configures certificate trust for Python, Node.js, and containerized applications without any additional configuration:
```
```suggestion
Aspire 13.0 automatically configures ASP.NET Development Certificate trust for Python, Node.js, and containerized applications without any additional configuration:
```

danegsta commented on tools/ReleaseNotes/data/whats-new-13.md:
````diff
@@ -0,0 +1,1561 @@
+    .WithDeveloperCertificateTrust(trust: true);
+```
+
+Features include loading from PEM files or certificate stores, flexible trust scoping, customizable container paths, and automatic dev certificate trust configuration.
````
Combine this with the `Certificate Trust Across Languages` section above as they're directly related features.

karolz-ms committed:
Explain how to enable container tunnel (#12813)

* Explain how to enable container tunnel

* Update container tunnel instructions

danegsta committed:
Update tools/ReleaseNotes/data/whats-new-13.md

danegsta committed:
Update tools/ReleaseNotes/data/whats-new-13.md

danegsta committed:
Cleanup entries on the lifecycle hook obsoletion (#12831)

danegsta committed:
Update tools/ReleaseNotes/data/whats-new-13.md

Commit:
Revise pipeline APIs note and CLI command examples

Updated the early preview note for pipeline APIs and clarified CLI command usage. Improved examples for custom pipeline steps and deployment commands.

davidfowl committed:
Add AppHost template updates for Aspire 13.0, simplifying project structure and SDK declaration

davidfowl committed:
Add support for single-file AppHosts in Aspire 13.0, simplifying setup for prototypes and learning scenarios

eerhardt committed:
eerhardt feedback

eerhardt committed:
More eerhardt feedback

jfversluis committed:
Enhance .NET MAUI integration details in release notes (#12872)

* Enhance .NET MAUI integration details in release notes

Updated the .NET MAUI integration section with new code examples and features for orchestrating mobile applications alongside cloud services.

* Update whats-new-13.md

---------

Co-authored-by: Mitch Denny <midenn@microsoft.com>

eerhardt committed:
Apply suggestion from @eerhardt

JamesNK committed:
Improve dashboard section

JamesNK committed:
Menu

JamesNK committed:
Menu

adamint committed:
add VS Code section, update vs code references

adamint committed:
add reference to vs code extension section
